tools/override-generator/main_test.go

```diff
@@ -0,0 +1,202 @@
+package main
+
+import (
+	"os"
+    "fmt"
+	"path/filepath"
+	"strings"
+	"testing"
+)
+
+func TestBzlmodOverride(t *testing.T) {
+	tests := []struct {
+		name string
+		give string
+		want string
+	}{
+		{
+			name: "simple no override",
+			give: `load("@bazel_gazelle//:deps.bzl", "go_repository")
+
+			go_repository(
+				name = "com_github_apache_thrift",
+				build_file_generation = "auto",
+				build_file_proto_mode = "default",
+				importpath = "github.com/apache/thrift",
+				sum = "h1:cMd2aj52n+8VoAtvSvLn4kDC3aZ6IAkBuqWQ2IDu7wo=",
+				version = "v0.17.0",
+			)`,
+			want: "",
+		},
+		{
+			name: "simple override",
+			give: `load("@bazel_gazelle//:deps.bzl", "go_repository")
+
+			go_repository(
+				name = "com_github_apache_thrift",
+				build_extra_args = ["-go_naming_convention_external=go_default_library"],
+				build_file_generation = "on",
+				build_file_proto_mode = "disable",
+				importpath = "github.com/apache/thrift",
+				sum = "h1:cMd2aj52n+8VoAtvSvLn4kDC3aZ6IAkBuqWQ2IDu7wo=",
+				version = "v0.17.0",
+			)`,
+			want: `go_deps.gazelle_override(
+				build_extra_args = ["-go_naming_convention_external=go_default_library"],
+				build_file_generation = "on",
+				directives = ["gazelle:proto disable"],
+				path = "github.com/apache/thrift",
+			)`,
+		},
+        {
+            name: "module override and gazelle",
+            give:  `load("@bazel_gazelle//:deps.bzl", "go_repository")
+
+            go_repository(
+                name = "com_github_bazelbuild_bazel_watcher",
+                build_extra_args = ["-go_naming_convention_external=go_default_library"],
+                build_file_generation = "off",  # keep
+                build_file_proto_mode = "disable",
+                importpath = "github.com/bazelbuild/bazel-watcher",
+                patch_args = ["-p1"],
+                patches = [
+                    # Remove it after they release this PR https://github.com/bazelbuild/bazel-watcher/pull/627
+                    "//patches:com_github_bazelbuild_bazel_watcher-go-embed.patch",
+                ],
+                sum = "h1:EfJzkMxJuNBGMVdEvkhiW7pAMwhaegbmAMaFCjLjyTw=",
+                version = "v0.23.7",
+            )`,
+            want: `go_deps.gazelle_override(
+                build_extra_args = ["-go_naming_convention_external=go_default_library"],
+                build_file_generation = "off",
+                directives = ["gazelle:proto disable"],
+                path = "github.com/bazelbuild/bazel-watcher",
+            )
+
+            go_deps.module_override(
+                patch_strip = 1,
+                patches = [
+                    #RemoveitaftertheyreleasethisPRhttps://github.com/bazelbuild/bazel-watcher/pull/627
```

Comment:
Are we losing spaces in comments?

Reply:
This was an issue with my test helper to try to compare the rules without spaces. I guess spaces are important in some situations, so I changed the helper to only trim leading and trailing spaces on each line.